packages/cli/src/util/dev/server.ts

```diff
@@ -1579,7 +1579,9 @@ export default class DevServer {
               const devServerParsed = new URL(this.address);
               if (devServerParsed.origin === rewriteUrlParsed.origin) {
                 // remove origin, leaving the path
-                req.url = rewritePath.slice(rewriteUrlParsed.origin.length);
+                req.url =
+                  rewritePath.slice(rewriteUrlParsed.origin.length) || '/';
```

Comment:
This is empty string in some cases. Default it to `/`.

Reply:
I'm fine with this, but I am curious couldn't you just do `req.url = rewriteUrlParsed.pathname + rewriteUrlParsed.search;`? No need to slice or worry about empty strings.

Reply:
I think the slice is actually safer because concatenation would also need to account for the port. Its really easy to get url parsing wrong.

Reply:
`port` is apart of the `origin`. `pathname` is everything after the port up to the query string.

Reply:
Ah right. Well then the missing piece is `hash` then

Reply:
I forgot about the `hash`, but given this is a server I believe the server never receives the `hash`.

Reply:
Yeah good point. I think either way is fine then 👍